contracts/nft_marketplace.ral

```diff
@@ -0,0 +1,98 @@
+import "nft_listing.ral"
+
+TxContract NFTMarketPlace(
+    nftListingByteCode: ByteVec,
+    mut admin: Address,
+    mut listingPrice: U256,
+    mut commissionRate: U256   // basis point. e.g. 2.5% == 250 basis point
+) {
+
+    event NFTListed(
+      price: U256,
+      tokenId: ByteVec,
+      tokenOwner: Address,
+      listingContractId: ByteVec,
+      listingContractAddress: Address
+    )
+
+    event AdminUpdated(
+        previous: Address,
+        new: Address
+    )
+
+    event ListingPriceUpdated(
+        previous: U256,
+        new: U256
+    )
+
+    event CommissionRateUpdated(
+        previous: U256,
+        new: U256
+    )
+
+    @using(preapprovedAssets = true, assetsInContract = true)
```

Comment:
It seems that `assetsInContract` is not necessary here.

Reply:
Seems to be a bug in the full node, where if we don't specify "assetsInContract = true" it will report `ContractAssetUnloaded`, investigating.

Reply:
As discussed, we are introducing a new method `payCommission` in `nft_marketplace.ral` to pay commission.

The problem is that if the contract asset is spent in an external call, the check of whether the contract asset is spent in the current method will not detect that. We solve that by move the code the pays the current contract (`nft_marketplace`) as a method inside the `nft_marketplace`.